autometa/__main__.py

```diff
@@ -0,0 +1,157 @@
+
+
+logger = logging.getLogger('autometa')
+
+
+def init_logger(fpath=None, level=logging.INFO):
+    """Initialize logger.
+
+    By default will initialize streaming logger with DEBUG level messages.
+    If `fpath` is provided, will write DEBUG level messages to `fpath` and
+    set streaming messages to INFO.
+
+    Parameters
+    ----------
+    fpath : str
+        </path/to/file.log>
+    level : int
+        Overwrite default logging level behavior with provided `level`.
+        This must be a constant from logging levels.
+        See https://docs.python.org/3/library/logging.html#levels for details.
+        i.e. logging.DEBUG, logging.INFO, etc. translates to 0,10, etc...
+
+    Returns
+    -------
+    logging.Logger
+        logging's Logger object to emit messages via methods:
+        'warn','info','debug','error','exception','critical','fatal'
+
+    Raises
+    -------
+    TypeError
+        `level` must be an int
+    ValueError
+        `level` must be one of 0, 10, 20, 30, 40, 50
+    """
+    levels = {
+        logging.NOTSET,
+        logging.DEBUG,
+        logging.INFO,
+        logging.WARNING,
+        logging.ERROR,
+        logging.CRITICAL}
+    if type(level) is not int:
+        raise TypeError(f'{level} must be an int! {type(level)}')
+    if level and level not in levels:
+        raise ValueError(f'{level} not in levels: {levels}!')
+    formatter = logging.Formatter(
+        fmt='[%(asctime)s %(levelname)s] %(name)s: %(message)s',
+        datefmt='%m/%d/%Y %I:%M:%S %p')
+    # Construct file/stream logging handlers
+    streamhandler = logging.StreamHandler()
+    streamhandler.setFormatter(formatter)
+    if fpath:
+        filehandler = logging.FileHandler(fpath)
+        filehandler.setFormatter(formatter)
+        logger.addHandler(filehandler)
+
+    streamhandler.setLevel(level)
+    logger.addHandler(streamhandler)
+    logger.setLevel(logging.DEBUG)
+    return logger
+
+def main(args):
+    # Setup logger
+    # timestamp = time.strftime("%Y-%m-%d_%H-%M-%S",time.gmtime())
+    # log_fpath = args.log if args.log else f'{timestamp}_autometa.log'
+    if args.debug:
+        logger = init_logger(fpath=args.log, level=logging.DEBUG)
+    else:
+        logger = init_logger(fpath=args.log)
+    # Configure AutometaUser
+    # TODO: master from WorkQueue is AutometaUser
+    user = AutometaUser(nproc=args.cpus)
+    user.configure(dryrun=args.check_dependencies)
+
+    for config in args.config:
+        # TODO: Add directions to master from WorkQueue
+        mgargs = user.prepare_binning_args(config)
+        user.run_binning(mgargs)
+        # user.refine_binning()
+        # user.process_binning()
+    # user.get_pangenomes()
+
+def entrypoint():
+    import argparse
+    import time
+    cpus = mp.cpu_count()
+    parser = argparse.ArgumentParser(description='Main script to run Autometa pipeline.')
+    parser.add_argument('config',
+        help='</path/to/metagenome.config>',
```

Comment:
As this is a user-facing help message, this should be more friendly. For example "Path to your config file". Also - if this is a positional argument, how do you specify more than one config file?

Reply:
You can pass in multiple just separate your inputs with a space and argparse collects this into a list.